packages/flutter_test/lib/src/window.dart

```diff
@@ -198,45 +208,51 @@ class TestWindow implements Window {
 
   @override
   double get textScaleFactor => _textScaleFactorTestValue ?? _window.textScaleFactor;
-  double _textScaleFactorTestValue;
+  double? _textScaleFactorTestValue;
   /// Hides the real text scale factor and reports the given
   /// [textScaleFactorTestValue] instead.
   set textScaleFactorTestValue(double textScaleFactorTestValue) {
     _textScaleFactorTestValue = textScaleFactorTestValue;
-    onTextScaleFactorChanged();
+    _testScaleFactorChanged();
   }
   /// Deletes any existing test text scale factor and returns to using the real
   /// text scale factor.
   void clearTextScaleFactorTestValue() {
     _textScaleFactorTestValue = null;
-    onTextScaleFactorChanged();
+    _testScaleFactorChanged();
   }
 
   @override
   Brightness get platformBrightness => _platformBrightnessTestValue ?? _window.platformBrightness;
-  Brightness _platformBrightnessTestValue;
+  Brightness? _platformBrightnessTestValue;
   @override
-  VoidCallback get onPlatformBrightnessChanged => _window.onPlatformBrightnessChanged;
+  VoidCallback? get onPlatformBrightnessChanged => _window.onPlatformBrightnessChanged;
   @override
-  set onPlatformBrightnessChanged(VoidCallback callback) {
+  set onPlatformBrightnessChanged(VoidCallback? callback) {
     _window.onPlatformBrightnessChanged = callback;
   }
   /// Hides the real text scale factor and reports the given
   /// [platformBrightnessTestValue] instead.
   set platformBrightnessTestValue(Brightness platformBrightnessTestValue) {
     _platformBrightnessTestValue = platformBrightnessTestValue;
-    onPlatformBrightnessChanged();
+    _platformBrightnessChanged();
   }
   /// Deletes any existing test platform brightness and returns to using the
   /// real platform brightness.
   void clearPlatformBrightnessTestValue() {
     _platformBrightnessTestValue = null;
-    onPlatformBrightnessChanged();
+    _platformBrightnessChanged();
+  }
+
+  void _platformBrightnessChanged() {
```

Comment:
All of these extra functions to test for nullness seem unnecessary to me.

Reply:
Removed them per your suggestion.